indra/llui/lltextbase.cpp

```diff
@@ -3233,6 +3261,24 @@ boost::signals2::connection LLTextBase::setIsObjectBlockedCallback(const is_bloc
 LLTextSegment::~LLTextSegment()
 {}
 
+// static
+LLStyle* LLTextSegment::cloneStyle(LLTextBase& target, const LLStyle* source)
+{
+    // Take most params from target
+    LLStyle::Params params = target.getStyleParams();
+    LLStyle* style = new LLStyle(params);
+
+    // Take some params from source
+    style->setLinkHREF(source->getLinkHREF());
+    if (source->isImage())
+    {
+        style->setImage(source->getImage()->getName());
+    }
+
+    return style;
```

Comment:
This whole idea of using `new` like this:
```
LLSomething* methodName()
{
    return new LLSomething();
}
```
without a corresponding `delete` looks like a huge potential for a memory leak. It would be much safer to use `LLPointer` instead.

Reply:
> It would be much safer to use `LLPointer` instead.

Done